### ISSUE TITLE
Code Generator (code_generator.py) Enhancements

### ISSUE THREAD

**Issue by Lemniscate-SHA-256** [Lemniscate-SHA-256/Neural-.nr#53]
Current Features:
- Generates TensorFlow and PyTorch models from parsed DSL files.
- Supports common layers like Conv2D, Dense, Dropout, Flatten.

 Next Enhancements:
- Dynamic Execution: Compile and train models directly from the DSL.
- Quantization & Pruning: Add TensorFlow Lite/PyTorch pruning for optimized inference.
- Custom Layers & Functions: Allow defining custom Python layers in the DSL.

**Comment by Lemniscate-SHA-256** [comment]
Fix for missing dataset in PyTorch training

`train_loader = torch.utils.data.DataLoader(MNIST(), batch_size=batch_size, shuffle=True)`
